Habicus/Habicus-Core-Web
# HABI-25 JPA Repo + Data Loader Updates and Service Implementation for Goals

![](https://camo.githubusercontent.com/cf373500dfe17773807191f7fb730ea7ea4b959d/68747470733a2f2f692e696d6775722e636f6d2f5878585a6b6d4f2e706e67)

## Ticket Reference ([View Current Issues Here](https://github.com/Habicus/Habicus-Core/issues))
My Issue Number Is: #25 


##### Core Reviewers:
@Habicus/core-team

##### Merge Owner:
@mweser
@Habicus/core-team
@austinsorrells
@schachte

## Status
**Ready**

## Description
- Added new API layer for goal controller
- Added new config for bean injection
- Updated models to support nested elements using foreign keys in hibernate
- Ability to custom query goals using DAO

tl;dr We can hit an endpoint and return goals associated with a user

## Steps to Test or Reproduce
1. Run server
2. Visit localhost:7777/api/v1/goal/{:id}
3. Verify results are present
